ykcapi/src/sginterp.rs

```diff
@@ -0,0 +1,142 @@
+use llvm_sys::bit_reader::LLVMParseBitcodeInContext2;
+use llvm_sys::core::*;
+use llvm_sys::prelude::{LLVMModuleRef, LLVMValueRef};
+use llvm_sys::{LLVMOpcode};
+use std::collections::HashMap;
+use std::ffi::CStr;
+use std::mem::MaybeUninit;
+
+use crate::llvmapihelper::{self};
+
+/// Stopgap interpreter values.
+#[derive(Debug)]
+pub enum SGValue {
```

Comment:
Are the only two types of possible value `U32` and `U64`?

Reply:
No, there'll be loads more.

Reply:
OK.